src/script/lua_api/l_object.cpp

```diff
@@ -2642,6 +2657,31 @@ int ObjectRef::l_set_lighting(lua_State *L)
 			lighting.bloom_radius          = getfloatfield_default(L, -1, "radius",          lighting.bloom_radius);
 		}
 		lua_pop(L, 1); // bloom
+
+		lua_getfield(L, 2, "vignette");
+		if (lua_istable(L, -1)) {
+			lighting.vignette.dark   = getfloatfield_default(L, -1, "dark",   lighting.vignette.dark);
+			lighting.vignette.bright = getfloatfield_default(L, -1, "bright", lighting.vignette.bright);
+			lighting.vignette.power  = getfloatfield_default(L, -1, "power",  lighting.vignette.power);
```

Comment:
```suggestion
			getfloatfield(L, -1, "dark",   lighting.vignette.dark);
			getfloatfield(L, -1, "bright", lighting.vignette.bright);
			getfloatfield(L, -1, "power",  lighting.vignette.power);
```

Reply:
It looks like most of the other code is formatted in this way, so why change it here specifically?

Reply:
1. It avoids redundancy, which can be a source of making mistakes.
2. Less code.

It's functionally identical so I'll leave the decision up to you.

Reply:
I agree, just I didn't want to change the code style. But I suppose it makes sense for new features to avoid redundancies, so I'll change it.